roles/ansible_service_broker/tasks/migrate.yml

```diff
@@ -0,0 +1,202 @@
+                    secret:
+                      secretName: asb-tls
+                  - name: asb-etcd-auth
+                    secret:
+                      secretName: broker-etcd-auth-secret
+                restartPolicy: Never
+                serviceAccount: asb
+                serviceAccountName: asb
+
+  - name: wait for migration to complete
+    oc_obj:
+      namespace: openshift-ansible-service-broker
+      kind: Job
+      state: list
+      name: asb-etcd-migration
+    register: migration_status
+    ignore_errors: true
+    until:
+      - "'results' in migration_status.results and migration_status.results.results | count > 0"
+      # Pod's 'Complete' status must be True
+      - "migration_status.results.results | lib_utils_oo_collect(attribute='status.conditions') | lib_utils_oo_collect(attribute='status', filters={'type': 'Complete'}) | map('bool') | select | list | count == 1"
+    delay: 10
+    retries: "{{ (asb_migration_timeout|default(600) | int / 10) | int }}"
+    failed_when:
+      - "'results' in migration_status.results"
+      - "migration_status.results.results | count > 0"
+      # Fail when pod's 'Failed' status is True
+      - "migration_status.results.results | lib_utils_oo_collect(attribute='status.conditions') | lib_utils_oo_collect(attribute='status', filters={'type': 'Failed'}) | map('bool') | select | list | count == 1"
+
+  - when: not (migration_status is failed)
+    block:
+      - name: Update broker configmap to use CRD backend
+        oc_obj:
+          name: broker-config
+          namespace: openshift-ansible-service-broker
+          state: present
+          kind: ConfigMap
+          content:
+            path: /tmp/cmout
+            data: "{{ lookup('template', 'configmap.yaml.j2') | from_yaml }}"
+        register: updated_configmap
+
+      - name: Update broker deploymentconfig
+        oc_obj:
+          force: yes
+          name: asb
+          namespace: openshift-ansible-service-broker
+          state: present
+          kind: DeploymentConfig
+          content:
+            path: /tmp/dcout
+            data: "{{ lookup('template', 'asb_dc.yaml.j2') | from_yaml }}"
+
+      - name: delete etcd service
+        oc_service:
+          name: asb-etcd
+          namespace: openshift-ansible-service-broker
+          state: absent
+
+      - name: delete etcd deploymentconfig
+        oc_obj:
+          name: asb-etcd
+          namespace: openshift-ansible-service-broker
+          kind: DeploymentConfig
+          state: absent
+
+      - name: delete broker etcd secret
+        oc_secret:
+          name: broker-etcd-auth-secret
+          namespace: openshift_ansible_service_broker
+          state: absent
+  always:
+    - name: scale up asb deploymentconfig
+      oc_scale:
+        name: asb
+        namespace: openshift-ansible-service-broker
+        kind: dc
+        replicas: 1
+
+- name: Fail out because the ASB etcd to CRD migration was unsuccessful
+  fail:
```

Comment:
If migration fails would this scale the broker back up with an etcd deployment? Or are we in a broken state? I see it's always scaling back up just unsure if the config is busted.

Reply:
if the migration fails it will still be on the etcd backend